File: Basics_with_JavaScript/910_For_Loop/More_Exersises/05._Game_Of_Intervals.js
function vladi(array) {
  let turnQty = Number(array[0]);
  let result = 0;
  let toTen = 0;
  let toTwenty = 0;
  let toThirty = 0;
  let toForty = 0;
  let toFifty = 0;
  let invalid = 0;

  for (let turn = 1; turn <= turnQty; turn++) {
    currentTurn = array[turn];

    if (currentTurn < 0 || currentTurn > 50) {
      invalid += 1;
      result /= 2;
      continue;
    } else if (currentTurn >= 0 && currentTurn <= 9) {
      result += currentTurn * 0.2;
      toTen += 1;
    } else if (currentTurn >= 10 && currentTurn <= 19) {
      result += currentTurn * 0.3;
      toTwenty += 1;
    } else if (currentTurn >= 20 && currentTurn <= 29) {
      result += currentTurn * 0.4;
      toThirty += 1;
    } else if (currentTurn >= 30 && currentTurn <= 39) {
      result += 50;
      toForty += 1;
    } else if (currentTurn >= 40 && currentTurn <= 50) {
      result += 100;
      toFifty += 1;
    }
  }

  console.log(result.toFixed(2));
  console.log(`From 0 to 9: ${((100 * toTen) / turnQty).toFixed(2)}%`);
  console.log(`From 10 to 19: ${((100 * toTwenty) / turnQty).toFixed(2)}%`);
  console.log(`From 20 to 29: ${((100 * toThirty) / turnQty).toFixed(2)}%`);
  console.log(`From 30 to 39: ${((100 * toForty) / turnQty).toFixed(2)}%`);
  console.log(`From 40 to 50: ${((100 * toFifty) / turnQty).toFixed(2)}%`);
  console.log(`Invalid numbers: ${((100 * invalid) / turnQty).toFixed(2)}%`);
}

vladi(["10", "43", "57", "-12", "23", "12", "0", "50", "40", "30", "20"]);

File: Basics_with_Python/PythonBasicCourse/4_For_Loop/10_For_Loop_Exercise/More_Exercise2/05._Game_Of_Intervals.py
result = 0
a = 0
b = 0
c = 0
d = 0
e = 0
f = 0
game_steps = int(input())
for _num in range(1, game_steps + 1):
    number = int(input())
    if 0 <= number <= 9:
        result += 0.20 * number
        a += 1
    elif 10 <= number <= 19:
        result += 0.30 * number
        b += 1
    elif 20 <= number <= 29:
        result += 0.40 * number
        c += 1
    elif 30 <= number <= 39:
        result += 50
        d += 1
    elif 40 <= number <= 50:
        result += 100
        e += 1
    elif number < 0 or number > 50:
        result /= 2
        f += 1
print(f"{result:.2f}")
print(f"From 0 to 9: {(100 * a / game_steps ):.2f}%")
print(f"From 10 to 19: {(100 * b / game_steps ):.2f}%")
print(f"From 20 to 29: {(100 * c / game_steps ):.2f}%")
print(f"From 30 to 39: {(100 * d / game_steps ):.2f}%")
print(f"From 40 to 50: {(100 * e / game_steps ):.2f}%")
print(f"Invalid numbers: {(100 * f / game_steps ):.2f}%")
